google-cloud-core/src/main/java/com/google/cloud/Policy.java

```diff
@@ -157,66 +174,147 @@ protected Builder(Policy policy) {
+          membersBuilder.add(identity.strValue());
+        }
+        bindingBuilder.setMembers(membersBuilder.build());
+        this.bindingsList.add(bindingBuilder.build());
+      }
+      return this;
+    }
+
+    /**
+     * Replaces the builder's List of bindings with the given List of Bindings.
+     *
+     * @throws NullPointerException if the given list is null, role is null, or contains any null
+     *     members in bindings.
+     */
+    public final Builder setBindings(List<Binding> bindings) {
+      this.bindingsList.clear();
+      for (Binding binding : bindings) {
+        Binding.Builder bindingBuilder = Binding.newBuilder();
+        bindingBuilder.setMembers(ImmutableList.copyOf(binding.getMembers()));
+        bindingBuilder.setRole(binding.getRole());
+        bindingBuilder.setCondition(binding.getCondition());
+        this.bindingsList.add(bindingBuilder.build());
       }
       return this;
     }
 
     /** Removes the role (and all identities associated with that role) from the policy. */
     public final Builder removeRole(Role role) {
-      bindings.remove(role);
+      checkArgument(
+          !isConditional(this.version, this.bindingsList),
+          "removeRole() is only supported with version 1 policies and non-conditional policies");
+      Iterator iterator = bindingsList.iterator();
+
+      while (iterator.hasNext()) {
+        Binding binding = (Binding) iterator.next();
+        if (binding.getRole().equals(role.getValue())) {
+          iterator.remove();
+          return this;
+        }
+      }
       return this;
     }
 
     /**
      * Adds one or more identities to the policy under the role specified.
      *
      * @throws NullPointerException if the role or any of the identities is null.
+     * @throws IllegalArgumentException if policy version is equal to 3 or has conditional bindings.
      */
     public final Builder addIdentity(Role role, Identity first, Identity... others) {
+      checkArgument(
+          !isConditional(this.version, this.bindingsList),
+          "addIdentity() is only supported with version 1 policies and non-conditional policies");
       String nullIdentityMessage = "Null identities are not permitted.";
       checkNotNull(first, nullIdentityMessage);
       checkNotNull(others, nullIdentityMessage);
-      for (Identity identity : others) {
-        checkNotNull(identity, nullIdentityMessage);
+      checkNotNull(role, "The role cannot be null.");
+      for (int i = 0; i < bindingsList.size(); i++) {
+        Binding binding = bindingsList.get(i);
+        if (binding.getRole().equals(role.getValue())) {
+          Binding.Builder bindingBuilder = binding.toBuilder();
+          ImmutableList.Builder<String> membersBuilder = ImmutableList.builder();
+          membersBuilder.addAll(binding.getMembers());
+          membersBuilder.add(first.strValue());
+          for (Identity identity : others) {
```

Comment:
```suggestion
          for (Identity identity : Lists.asList(first, others)) {
```

And then you don't need the previous line.

Reply:
This is necessary due to type transformation.